lib/Directives/SubDirective.php

```diff
@@ -30,10 +31,44 @@ final public function process(
         string $data,
         array $options
     ): void {
-        $subParser = $parser->getSubParser();
+        if ($node === null) {
+            /*
+             * A directive with no content (which is not allowed for sub-directives.
+             *
+             * For example:
+             *
+             *      .. note::
+             *
+             *      Now more normal text.
+             */
+            throw new RuntimeException('Content expected, none found.');
+        }
 
-        if ($node instanceof CodeNode) {
-            $document = $subParser->parseLocal($node->getValue());
+        /*
+         * A BlockNode indicates that the content was passed in "raw" and should
+         * be sub-parsed. This is the main use-case.
+         *
+         * However, there are some odd cases where a directive appears
+         * to "end"... and DocumentParser continues parsing the next
+         * section, for example as a ParagraphNode or ListNode. When
+         * that terminates, the original directive is THEN processed
+         * and called.
+         *
+         * A key example is the "class::" directive (see class-directive.rst test case).
+         * That is where it is legal to have a format like this:
+         *
+         *      .. class:: special-list
+         *
+         *      - Test list item 1.
+         *      - Test list item 2.
+         *
+         * Notice the 2 list items are NOT indented. This is legal, and ultimately
+         * those two items would be parsed as a ListNode and THEN passed to
+         * ClassDirective (which extends SubDirective) for processing.
+         */
+        if ($node instanceof BlockNode) {
```

Comment:
Switching `wantCode()` from `true` to `false` basically just mean that in the "normal" situation, the `CodeNode` became a `BlockNode`. So the logic here is the same as before... but I added a bunch of docs to describe why there is/was this if statement.